pkg/gardenlet/controller/shoot/shoot_control.go

```diff
@@ -367,6 +383,19 @@ func (c *Controller) finalizeShootDeletion(ctx context.Context, logger *logrus.E
 	return reconcile.Result{}, c.removeFinalizerFrom(shoot)
 }
 
+func (c *Controller) isSeedAvailable(seed *gardencorev1beta1.Seed, logger *logrus.Entry) bool {
+	if seed.DeletionTimestamp != nil {
+		return false
+	}
+
+	if err := health.CheckSeed(seed, c.identity); err != nil {
+		logger.Debugf("seed is not yet ready: %v", err)
+		return false
+	}
+
+	return true
```

Comment:
Imo: No need to log this here (especially on debug level) as the status will be returned in form of an error anyways.

```suggestion
	return health.CheckSeed(seed, c.identity)
```